index.bs

```diff
@@ -2581,6 +2665,62 @@ an optional [=boolean=] |isNoised| (default false):
+
+To <dfn>remove attribution scopes for sources with destination</dfn> given a [=site=] |destination| and an [=attribution source=] |pendingSource|:
+1. Let |attributionScopeRecords| be a new [=list=].
+1. [=set/iterate|For each=] |attributionScope| in |pendingSource|'s [=attribution source/attribution scopes=]:
+    1. [=list/Append=] the [=tuple=] (|attributionScope|, |pendingSource|'s [=attribution source/source time=], |pendingSource|'s [=attribution source/source identifier=]) to |attributionScopeRecords|.
+1. [=set/iterate|For each=] |source| of the [=attribution source cache=]:
+    1. If |source|'s [=attribution source/reporting origin=] and |pendingSource|'s [=attribution source/reporting origin=] are not [=same origin=], [=iteration/continue=].
+    1. If |source|'s [=attribution source/attribution destinations=] does not [=set/contain=] |destination|, [=iteration/continue=].
+    1. If |source|'s [=attribution source/attribution scopes=] [=set/is empty=], [=iteration/continue=].
+    1. [=set/iterate|For each=] |attributionScope| in |source|'s [=attribution source/attribution scopes=]:
+        1. [=list/Append=] the [=tuple=] (|attributionScope|, |source|'s [=attribution source/source time=], |source|'s [=attribution source/source identifier=]) to |attributionScopeRecords|.
+1. [=list/sort in ascending order|Sort=] |attributionScopeRecords| in ascending order with |a| being less than |b| if any of the following are true:
+      * |a|[1] is greater than |b|[1].
+      * |a|[1] is equal to |b|[1] and |a|[0] is greater than |b|[0].
+1. Let |selectedAttributionScopes| be a new [=set=].
+1. Let |sourceIdsToClearAttributionScopes| be a new [=set=].
+1. [=list/iterate|For each=] |record| of |attributionScopeRecords|:
+    1. If |selectedAttributionScopes|'s [=set/size=] is less than |pendingSource|'s [=attribution source/attribution scope limit=], [=set/append=] |record|[0] to |selectedAttributionScopes|.
+    1. Otherwise, if |selectedAttributionScopes| does not [=set/contain=] |record|[0], [=set/append=] |record|[2] to |sourceIdsToClearAttributionScopes|.
+1. [=set/iterate|For each=] |source| of the [=attribution source cache=]:
+    1. If |sourceIdsToClearAttributionScopes| [=set/contains=] |source|'s [=attribution source/source identifier=], set |source|'s [=attribution source/attribution scopes=] to a new [=set=].
+
+To <dfn>remove attribution scopes for sources with attribution scopes not selected if necessary</dfn> given an [=attribution source=] |pendingSource|:
+1. Let |attributionScopeLimit| be |pendingSource|'s [=attribution source/attribution scope limit=].
+1. If |attributionScopeLimit| is null, return.
+1. [=set/iterate|For each=] |destination| in |pendingSource|'s [=attribution source/attribution destinations=]:
+    1. [=Remove attribution scopes for sources with destination=] with |attributionScopeLimit|, |destination|, and |pendingSource|'s [=attribution source/reporting origin=].
```

Comment:
This calls a method with three parameters, but the method, "Remove attribution scopes for sources with destination," is defined with only two parameters.